src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs

```diff
@@ -21,7 +21,9 @@ public static class TextLoaderSaverCatalog
         /// <param name="catalog">The <see cref="DataOperationsCatalog"/> catalog.</param>
         /// <param name="columns">Array of columns <see cref="TextLoader.Column"/> defining the schema.</param>
         /// <param name="separatorChar">The character used as separator between data points in a row. By default the tab character is used as separator.</param>
-        /// <param name="hasHeader">Whether the file has a header.</param>
+        /// <param name="hasHeader">Whether the file has a header with feature names. Note: If a TextLoader is created with HasHeader=true but without a dataSample,
+        /// then the TextLoader will not contain slot (columns that are chosen for manipulation) names, because the output schema is made when the TextLoader is made,
```

Comment:
>columns that are chosen for manipulation [](start = 55, length = 40)

the term "slots" is how we refer to the elements of a vector column. In `TextLoader`, these correspond to actual columns (in the "Excel" sense) in the text file. For example, if a user defines `TextLoader.Column("Features", 3, 14)`, that would create a vector column called `Features`, with 12 slots, and values taken from columns 3 thru 14 in the text file. If the text loader is given a `dataSample` and `hasHeader` is true, then the column `Features` will have slot names annotations, with values that will be taken from the first line, columns 3 thru 14 in the `dataSample` file.

Reply:
Got it, I've updated the summary here.

Reply:
The comment in the parentheses here (and in the other public APIs) also needs to be updated.

---
In reply to: [370074523](https://github.com/dotnet/machinelearning/pull/4655#discussion_r370074523) [](ancestors = 370074523)

Reply:
Can you add the part about skipping the first line to the documentation in this file as well? Thanks.

---
In reply to: [370987713](https://github.com/dotnet/machinelearning/pull/4655#discussion_r370987713) [](ancestors = 370987713,370074523)

Reply:
Hi Yael, done and done!